packages/api/src/v2/resourcePlugin.ts

```diff
@@ -114,17 +114,15 @@ export interface ResourcePlugin {
    *
    * @param {Context} ctx - plugin's runtime context shared by all lifecycles.
    * @param {ProvisionInputs} inputs - inputs injected by Toolkit runtime and solution.
-   * @param {DeepReadonly<EnvInfoV2>} envInfo - a readonly view of environment info modeled after (config|state).${env}.json
+   * @param {EnvInfoV2} envInfo - a readonly view of environment info modeled after (config|state).${env}.json
```

Comment:
Curious why the env info no longer need to be readonly?

Reply:
Preview API, each plugin will return provisioned output, which will make the API v2 more complicated.
From practice, EnvInfoV2 is kept as both input and output (by reference) is more easy to work than value copy from return